watertap/examples/flowsheets/case_studies/activated_sludge/modified_ASM2D_flowsheet.py

```diff
@@ -0,0 +1,388 @@
+    m.fs.R3.volume.fix(1000 * pyo.units.m**3)
+    m.fs.R4.volume.fix(1000 * pyo.units.m**3)
+    m.fs.R5.volume.fix(1333 * pyo.units.m**3)
+    m.fs.R6.volume.fix(1333 * pyo.units.m**3)
+    m.fs.R7.volume.fix(1333 * pyo.units.m**3)
+
+    # Injection rates to Reactions 3, 4 and 5
+    for j in m.fs.props.component_list:
+        if j != "S_O2":
+            # All components except S_O have no injection
+            m.fs.R5.injection[:, :, j].fix(0)
+            m.fs.R6.injection[:, :, j].fix(0)
+            m.fs.R7.injection[:, :, j].fix(0)
+    # Then set injections rates for O2
+    m.fs.R5.outlet.conc_mass_comp[:, "S_O2"].fix(1.91e-3)
+    m.fs.R6.outlet.conc_mass_comp[:, "S_O2"].fix(2.60e-3)
+    m.fs.R7.outlet.conc_mass_comp[:, "S_O2"].fix(3.20e-3)
+
+    # Set fraction of outflow from reactor 5 that goes to recycle
+    m.fs.SP1.split_fraction[:, "underflow"].fix(0.60)
+
+    # Clarifier
+    # TODO: Update once more detailed model available
+    m.fs.CL1.split_fraction[0, "effluent", "H2O"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_A"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_F"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_I"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_N2"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_NH4"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_NO3"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_O2"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_PO4"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_IC"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_K"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_Mg"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "X_AUT"].fix(0.022117)
+    m.fs.CL1.split_fraction[0, "effluent", "X_H"].fix(0.021922)
+    m.fs.CL1.split_fraction[0, "effluent", "X_I"].fix(0.021715)
+    m.fs.CL1.split_fraction[0, "effluent", "X_PAO"].fix(0.022)
+    m.fs.CL1.split_fraction[0, "effluent", "X_PHA"].fix(0.02147)
+    m.fs.CL1.split_fraction[0, "effluent", "X_PP"].fix(0.02144)
+    m.fs.CL1.split_fraction[0, "effluent", "X_S"].fix(0.02221)
+
+    # Sludge purge separator
+    m.fs.SP2.split_fraction[:, "recycle"].fix(0.97955)
+
+    # Outlet pressure from recycle pump
+    m.fs.P1.outlet.pressure.fix(101325)
+
+    # Check degrees of freedom
+    print(f"DOF after all: {degrees_of_freedom(m)}")
+    assert degrees_of_freedom(m) == 0
+
+    def scale_variables(m):
+        for var in m.fs.component_data_objects(pyo.Var, descend_into=True):
+            if "flow_vol" in var.name:
+                iscale.set_scaling_factor(var, 1e1)
+            if "temperature" in var.name:
+                iscale.set_scaling_factor(var, 1e-1)
+            if "pressure" in var.name:
+                iscale.set_scaling_factor(var, 1e-4)
+            if "conc_mass_comp" in var.name:
+                iscale.set_scaling_factor(var, 1e1)
```

Comment:
does flowsheet fail to converge if you take these away?

Reply:
Yeah, the flowsheet will fail to converge if I don't set these scaling factors